daemon/api_validate.go

```diff
@@ -114,9 +114,6 @@ func listValidationSets(c *Command, r *http.Request, _ *auth.UserState) Response
 	for i, vs := range names {
 		tr := validationSets[vs]
 		sequence := tr.Current
-		if tr.PinnedAt > 0 {
-			sequence = tr.PinnedAt
-		}
 		sets, err := validationSetForAssert(st, tr.AccountID, tr.Name, sequence)
```

Comment:
we can't do this because there is problematic code in assertstate that violates this. Indeed is that code that I think we should remove first.  Regarding this my preference might be to add a Sequence() method ValidationSetTracking that does this and have other places using that

Reply:
I would still recommend to make this change as well as turning:

validationSetForAssert(st *state.State, accountID, name string, sequence int) => validationSetsForTracked(*state.State, *assertstate.ValidationSetTracking)

Reply:
Sounds good :+1: